components/layout/Nav/ItemLink.js

```diff
@@ -1,16 +1,53 @@
-import { Link } from 'react-router-dom';
+import { NavLink } from 'react-router-dom';
+import { useState } from 'react';
+
+const getLink = (parentProps, props) => (
+    <a href={props.href} className={parentProps.className}>
+        {parentProps.children}
+    </a>
+);
+
+const getListItem = (parentProps, props) => {
+    const [focus, setFocus] = useState(false);
+
+    const dropDownClasses = `${
+        props.className
+    } rd-navbar--has-megamenu rd-navbar-submenu ${focus ? 'focus' : ''}`;
 
-const ItemLink = props => {
     return (
-        <Link
-            className={props.className}
-            to={{
-                pathname: props.pathname,
-                state: props.state
+        <li
+            className={
+                parentProps.topMenu && parentProps.dropDown ? dropDownClasses : props.className
+            }
+            onMouseOver={() => {
+                setFocus(true);
+            }}
+            onMouseOut={() => {
+                setFocus(false);
             }}
         >
-            {props.children}
-        </Link>
+            {!Array.isArray(parentProps.children) ? (
+                getLink(parentProps, props)
+            ) : (
+                <>{parentProps.children}</>
+            )}
+        </li>
+    );
+};
+
+const ItemLink = parentProps => {
+    return (
+        <NavLink
+            component={props => {
+                return parentProps.topMenu ? getListItem(parentProps, props) : getLink(parentProps, props);
```

Comment:
In react, this should be components

```suggestion
                return parentProps.topMenu ? <ListItem {...parentProps, ...props} /> : <Link {...parentProps, ...props} />;
```